python/tk_hiero_export/version_creator.py

```diff
@@ -128,6 +129,27 @@ def __init__(self, initDict):
         self._quicktime_path = None
         self._temp_quicktime = None
 
+    def addWriteNodeToScript(self, script, rootNode, framerate):
+        """
+        Override the default addWriteNodeToScript functionality so that we can add a SetNode before
+        the default writenodes get added to the script. The SetNode will allow us to later push our mov writenode
+        to this point so that it get parented correctly.
+        :param script:
+        :param rootNode:
+        :param framerate:
+        :return:
+        """
+        self.app.log_debug("Adding SetNode before base write node gets added")
+        # We create and store an arbitrary name for our SetNode, this is used as an id, so that when we create
+        # the PushNode later we can connect it to this SetNode.
+        self.write_set_node_label = "SG_Write_Attachment"
+
+        # Add the SetNode before the write nodes are added.
+        setCommand = nuke.SetNode(self.write_set_node_label, 0)
+        script.addNode(setCommand)
+
+        super(ShotgunTranscodeExporter, self).addWriteNodeToScript(script, rootNode, framerate)
```

Comment:
Since you're not using `script`, `rootNode`, and `framerate` for anything other than to just pass them through, skip that entirely and just use `*args, **kwargs` instead. That'll also protect us from them changing the interface to this method in the future since we'll just be passing through whatever they give us.

```
def addWriteNodeToScript(self, *args, **kwargs):
    ...
    super(ShotgunTranscodeExporter, self).addWriteNodeToScript(*args, **kwargs)
```

Reply:
Yes good point!

Reply:
ah spoke too soon, I am using script.